src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -797,17 +797,19 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     // but using the iterator instead.
 
     const iteratorFn = getIteratorFn(newChildrenIterable);
-    if (typeof iteratorFn !== 'function') {
-      throw new Error('An object is not an iterable.');
-    }
+    invariant(
+      typeof iteratorFn === 'function',
+      'An object is not an iterable.'
+    );
 
     if (__DEV__) {
       // First, validate keys.
       // We'll get a different iterator later for the main pass.
       const newChildren = iteratorFn.call(newChildrenIterable);
-      if (newChildren == null) {
-        throw new Error('An iterable object provided no iterator.');
-      }
+      invariant(
```

Comment:
This isn't a behavior change in your PR, but we shouldn't have any DEV-only exceptions. Can we make this a warning (either here in a new PR)?